vector/src/main/java/im/vector/app/features/notifications/NotificationEventPersistence.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.notifications
+
+import android.content.Context
+import org.matrix.android.sdk.api.session.Session
+import timber.log.Timber
+import java.io.File
+import java.io.FileOutputStream
+
+// TODO Multi-account
+private const val ROOMS_NOTIFICATIONS_FILE_NAME = "im.vector.notifications.cache"
+private const val KEY_ALIAS_SECRET_STORAGE = "notificationMgr"
+
+object NotificationEventPersistence {
```

Comment:
Is is possible to inject the constructor with the context to simplify the API?

Reply:
will do :+1: it will mean passing the `persistence` into the `NotificationState` methods that use it or lifting those interactions back to the `NotificationManagerDrawer`

Reply:
done https://github.com/vector-im/element-android/pull/4506/commits/7693f3bc6c7607ade8015838e0cbfaf3bdc1160f